app/util/FeatureFlagSupport.scala

```diff
@@ -23,7 +23,13 @@ import play.api.mvc.Results.Gone
 trait FeatureFlagSupport {
 
   def ifPartialsEnabled(block: Result)(implicit appConfig: AppConfig): Result =
-    if (appConfig.disablePartials) {
+    ifEnabled(block, appConfig.disablePartials)
+
+  def ifAjaxPartialsEnabled(block: Result)(implicit appConfig: AppConfig): Result =
+    ifEnabled(block, appConfig.disableAjaxPartials)
+
+  private def ifEnabled(block: Result, isDisabled: Boolean): Result =
```

Comment:
@JoPintoPaul just spotted, does this need to be pass by name https://docs.scala-lang.org/tour/by-name-parameters.html like `block: => Result` so that we aren't calculating the Result block regardless of the feature flag

Reply:
guess on each of the functions in the chain